### PR TITLE
Unify types of all 'style' related props in AnimateProps

### PR DESCRIPTION
**Pull request by tjzel**
## Summary

This pull requests is a simple rewrite of #3653 since we will no longer have the .d.ts file.

I've also fixed `react-native-reanimated-tests.tsx` so it would be able to resolve its imports properly.

## Test plan

`yarn type:check`, `yarn type:check-api`, also a look into new tests in `react-native-reanimated-tests.tsx`.

---

Co-authored by @johankasperi

### CODE REVIEW

File: __typetests__/react-native-reanimated-tests.tsx [hunk 1]
```diff
@@ -57,10 +57,9 @@ import Animated, {
   StyleLayoutAnimation,
   Animation,
   // eslint-disable-next-line import/no-unresolved
-} from 'react-native-reanimated';
+} from '../src';
```
Comment (tomekzaw) [inline __typetests__/react-native-reanimated-tests.tsx:60]: I'm not a huge fan of this change, `import { ... } from 'react-native-reanimated';` also checks if there's a module called `react-native-reanimated`. Why is it necessary?

File: __typetests__/react-native-reanimated-tests.tsx [hunk 2]
```diff
@@ -57,10 +57,9 @@ import Animated, {
...
 import {
   dispatchCommand,
-  getTag,
```
Comment (tomekzaw) [inline __typetests__/react-native-reanimated-tests.tsx:63]: This change is necessary because I've removed `getTag` function in #4361.

**Review by johankasperi — APPROVED** [review]
@tjzel Looks good! And thanks for picking up my original PR and transform it so it matches the relevant files. I definitely feel like I have gotten enough credit 😄 Thanks again!

File: __typetests__/react-native-reanimated-tests.tsx (hunk 1 above)
Comment (tjzel) [reply __typetests__/react-native-reanimated-tests.tsx:60]: In my environment TypeScript couldn't resolve this import properly and inferred all of the imports as `any`, making the tests kind-of useless. I'd be more than happy if you could check if that's the case for you too.

While this file can check if there's a module called `react-native-reanimated`, this file isn't considered anywhere in our CIs etc., hence even if there was something wrong there we wouldn't know until we've had it manually checked. I think this file should focus on types only - and if we want, we could do much more reliable test that checks if module called `react-native-reanimated` exists.

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**